src/util/fs.ts

```diff
@@ -172,5 +175,74 @@ export const fs = {
     } catch (err) {
       return false;
     }
+  },
+
+  /**
+   * Recursively act on all files or directories in a directory
+   *
+   * @param dir path to directory
+   * @param perform function to be run on contents of dir
+   * @param onType optional parameter to specify type to actOn
+   * @returns void
+   */
+
+  actOn: async (dir: string, perform: PerformFunction, onType?: string): Promise<void> => {
+    // Act on files by default
+    onType = onType || 'file';
+
+    for (const file of await fs.readdir(dir)) {
+      const filePath = path.join(dir, file);
+      const stat = await fs.stat(filePath);
+
+      if (stat) {
+        if (stat.isDirectory()) {
+          await fs.actOn(filePath, perform, onType);
+          if (onType === 'dir') {
+            await perform(filePath);
+          }
+        } else if (stat.isFile() && onType === 'file') {
+          await perform(filePath, file, dir);
+        }
+      }
+    }
+  },
+
+  /**
+   * Checks if files are the same
+   *
+   * @param file1Path the first file path to check
+   * @param file2Path the second file path to check
+   * @returns boolean
+   */
+  areFilesEqual: async (file1Path: string, file2Path: string): Promise<boolean> => {
+    let contentA: string;
+    let contentB: string;
+
+    try {
+      contentA = await fs.readFile(file1Path, 'utf8');
+    } catch (error) {
+      throw new SfdxError(`The path: ${file1Path} doesn\'t exist.`, 'DirMissingOrNoAccess');
+    }
+    try {
+      contentB = await fs.readFile(file2Path, 'utf8');
+    } catch (error) {
+      throw new SfdxError(`The path: ${file1Path} doesn\'t exist.`, 'DirMissingOrNoAccess');
+    }
```

Comment:
Can we combine these two try/catch? I bet somewhere in the error you can find the filepath, also the second try/catch should have the message `The path: ${file2Path} doesn\'t exist.`, 'DirMissingOrNoAccess' - you have file1path

Reply:
Yep, good call!